src/PROPOSAL/PROPOSAL/secondaries/parametrization/photopairproduction/PhotoPairProductionKochMotz.h

```diff
@@ -15,5 +15,43 @@ namespace PROPOSAL {
             {}
 
         };
+
+        class PhotoPairProductionKochMotzSauter : public PhotoPairProductionKochMotzForwardPeaked {
+        public:
+            PhotoPairProductionKochMotzSauter() = default;
+            PhotoPairProductionKochMotzSauter(ParticleDef p, Medium m)
+                : PhotoPairProductionKochMotzForwardPeaked(p, m)
+            {}
+
+            std::tuple<Cartesian3D, Cartesian3D> CalculateDirections(
+                    const Vector3D& dir, double energy, double rho,
+                    const Component& comp, double rnd1, double rnd2, double rnd3) override {
+
+                // Sampling according to EGS 5 manual, formula (2.157)
+
+                auto energies = CalculateEnergy(energy, rho);
+                double E_electron = std::get<0>(energies);
+                double E_positron = std::get<1>(energies);
+                double p_electron = std::sqrt(E_electron * E_electron - ME * ME);
+                double p_positron = std::sqrt(E_positron * E_positron - ME * ME);
+
+                // TODO: should this be two times the same random numbers, or two different ones?
```

Comment:
Have you tried both and did you see a bias when using the same rnd?
I would personally say, that this shouldn't be independent.

Reply:
I've had a look at the source code of EGS5, and there, they use two independent random numbers for the sampling procedure.

(For the record: I found this in `egs5_pair.f`, lines 95 and following.)

Reply:
ok, then let|s make it consistent with EGS and leave it like this